src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -90,9 +130,17 @@ public void RegisterMissingWorkloads(Guid projectGuid, ConfiguredProject project
             UnregisterProjectConfiguration(projectGuid, project);
         }
 
-        public void RegisterMissingSdkRuntimeComponentId(Guid projectGuid, ConfiguredProject project, string runtimeComponentId)
+        public void RegisterPossibleMissingSdkRuntimeVersion(Guid projectGuid, ConfiguredProject project, string runtimeVersion)
         {
-            _projectGuidToRuntimeDescriptorMap.GetOrAdd(projectGuid, runtimeComponentId);
+            // Workaround to fix #AB1460328.
+            // VS has no information about the packages installed outside VS, and deep detection is not suggested for performance reasons.
+            // This workaround reads the Registry Key HKLM\SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\sharedfx\Microsoft.NETCore.App
+            // and get the installed runtime versions from the value names.
+            if (!string.IsNullOrEmpty(runtimeVersion) && !NetCoreRegistryKeyValues.Contains(runtimeVersion))
+            {
+                s_packageVersionToComponentId.TryGetValue(runtimeVersion, out string? componentId);
+                _projectGuidToRuntimeDescriptorMap.GetOrAdd(projectGuid, componentId!);
```

Comment:
Is `componentId` guaranteed to not be null here (I'm looking at that exclamation point)? `TryGetValue` should be part of the if-statement because if you don't get a value, what is the purpose of running `GetOrAdd`?